### PR TITLE
typing: various utils

### PR DESCRIPTION
**Pull request by tdesveaux**
Type:
- RemoteCommand
- unicode2bytes
- bytes2unicode
- LineBoundaryFinder
- deferredLocked

## Contributor Checklist:

* [n/a] I have updated the unit tests
* [n/a] I have created a file in the `newsfragments` directory (and read the `README.txt` in that directory)
* [n/a] I have updated the appropriate documentation

### CODE REVIEW

File: master/buildbot/process/remotecommand.py [hunk 1]
```diff
@@ -51,61 +59,75 @@ class RemoteCommand(base.RemoteCommandImpl):
...
+        self._closeWhenFinished: dict[str, bool] = {}
+        self.collectStdout: bool = collectStdout
+        self.collectStderr: bool = collectStderr
         self.stdout = ''
         self.stderr = ''
-        self.updates = {}
-        self.stdioLogName = stdioLogName
-        self._startTime = None
-        self._remoteElapsed = None
+        self.updates: defaultdict[str, list[Any]] = defaultdict(list)
```
Comment (p12tic) [inline master/buildbot/process/remotecommand.py:85]: This is not just type change, best would be to split it into separate commit. Same with line boundary finders.

Comment (tdesveaux) [reply master/buildbot/process/remotecommand.py:85]: Indeed, missed it, sorry about that.

**Review by p12tic — APPROVED** [review]
